### PR TITLE
Add edit, delete, add actions directly to the affirmation card

### PR DESCRIPTION
**Pull request by Shaomei**
This PR implements the following functionalities for the affirmation messages.

1. Add edit and delete actions directly to the affirmation card.
2. Add a "+" button in the end of affirmation carousal to add a new card.

With these functionalities directly implemented in affirmation carousal, we are also able to remove the affirmation tab and states that were previous used for the affirmation tab. 

I also removed the hand wave tab since we don't need it for now. 

Try out [here](https://humpback-immune-frankly.ngrok-free.app).

See demo video [here](https://drive.google.com/file/d/1jYC5Ow0bzwZF6313DXdZDG6vTZXhVbat/view?usp=sharing).

### CODE REVIEW

File: app/main/page.tsx [hunk 1]
```diff
@@ -36,19 +34,14 @@ const defaultWaveHandButtons = [
 ];
 
 const defaultAffirmations = [
-    'Say what I want to say, whatever happens will help me grow',
-    'I can take up space',
-    'I have an important voice',
-    'Feel the tension and proceed',
-    'I have the right to stutter',
-]
+    {id: 0, text: 'Say what I want to say, whatever happens will help me grow' },
```
Comment (blickly) [inline app/main/page.tsx:37]: Do we already need ids even when this isn't connected to the DB?

File: components/EditContentMenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    </React.Fragment>
+  );
+}
+
+interface DeleteContentMenuItemProps {
+  id: number;
+  deleteCard: (id: number) => void;
+}
+
+export function DeleteContentMenuItem(
```
[the remark this answers is not in the document]
Comment (Shaomei) [reply components/EditContentMenuItem.tsx]: Done. I created a new file components/DeleteContentMenuItem.tsx.

File: app/main/page.tsx (hunk 1 above)
Comment (Shaomei) [reply app/main/page.tsx:37]: It seems to be the easiest way to find a replace the value of one item in the list. 

Set does not seem to support single item update.. 

See some examples for array manipulation [here](https://react.dev/learn/updating-arrays-in-state).

File: components/DeleteContentMenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import MenuItem from "@mui/material/MenuItem";
+import Modal from "@mui/material/Modal";
+import '@/app/css/Affirmation.css';
+
+interface DeleteContentMenuItemProps {
+  id: number;
+  deleteCard: (id: number) => void;
```
Comment (blickly) [inline components/DeleteContentMenuItem.tsx:8]: By convention, these callback parameters to HTML elements (and I assume to react components as well) usually start with "on", like "onClick", "onError", "onUsernameChanged", etc.)

File: components/EditContentMenuItem.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import MenuItem from "@mui/material/MenuItem";
+import Modal from "@mui/material/Modal";
+import '@/app/css/Affirmation.css';
+
+interface EditContentMenuItemProps {
+  id: number;
+  initialText: string;
+  updateContent: (id: number, updatedText: string) => void;
```
Comment (blickly) [inline components/EditContentMenuItem.tsx:9]: This should also start with "on", I think

**Review by blickly — APPROVED** [review]
LGTM modulo a few parameter names

File: components/WriteAffirmationCardModal.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import Modal from "@mui/material/Modal";
+import '@/app/css/Affirmation.css';
+
+interface WriteAffirmationCardModalProps {
+  open: boolean;
+  handleModalClose: () => void;
```
Comment (blickly) [inline components/WriteAffirmationCardModal.tsx:7]: This can also be "on" instead of "handle" I think

File: components/AffirmationCard.tsx [hunk 6]
```diff
@@ -1,15 +1,92 @@
...
+import '@/app/css/Affirmation.css';
+
+export interface AffirmationCardContent {
+  id: number;
+  text: string;
+}
 
 interface AffirmationCardProps {
- text: string;
+ currentCard: AffirmationCardContent;
```
Comment (blickly) [inline components/AffirmationCard.tsx:22]: initialContent?

File: __tests__/affirmation.test.tsx [hunk 7]
```diff
@@ -1,129 +0,0 @@
...
-
-    const dropdownButton = screen.getByText('Affirmation 1');
-    fireEvent.click(dropdownButton);
-    const dropdown = within(dropdownButton.nextSibling);
-    const deleteButton = dropdown.getByText('Delete');
-    fireEvent.click(deleteButton);
-
-    expect(screen.queryByText('Affirmation 1')).not.toBeInTheDocument();
-    
-  });
```
Comment (blickly) [inline __tests__/affirmation.test.tsx:103]: Test that edit and delete do something seems useful.  What's the plan?

**Review by blickly — APPROVED** [review]

File: __tests__/affirmation.test.tsx (hunk 7 above)
Comment (Shaomei) [reply __tests__/affirmation.test.tsx:103]: See issue #77

File: components/EditContentMenuItem.tsx (hunk 4 above)
Comment (Shaomei) [reply components/EditContentMenuItem.tsx]: Done

File: components/DeleteContentMenuItem.tsx (hunk 3 above)
Comment (Shaomei) [reply components/DeleteContentMenuItem.tsx]: Done.

File: components/AffirmationCard.tsx (hunk 6 above)
Comment (Shaomei) [reply components/AffirmationCard.tsx:22]: Done.

File: components/WriteAffirmationCardModal.tsx (hunk 5 above)
Comment (Shaomei) [reply components/WriteAffirmationCardModal.tsx:7]: Done.

**Pull request merged**